### PR TITLE
feat: add custom primary actions for wizard component

### PR DESCRIPTION
**Pull request by SpyZzey**
### Description

Adds functionality to overwrite the content of the primary actions for wizards (e.g. Next, Previous, Cancel, ...).
Currently, the Wizard only allows the default primary actions (Next, Previous, Cancel, Finish, Skip to ...). To allow further flexibility, a `customPrimaryActions`-property was introduced to allow custom actions.

Related links, issue #, if available: n/a

### How has this been tested?

**Unit Testing**
- Added unit test to check if custom primary actions are rendered instead of default buttons
- Added unit test to check if custom primary actions work on all steps
- Added unit test to check if custom primary actions overwrite skip-to button
- Added unit test to check if default buttons are used if custom primary actions are null or undefined

**Manual Testing**
- Confirmed custom primary actions are being rendered
- Confirmed custom primary actions can be used to navigate steps
- Confirmed custom primary actions can be customized per step

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by just-boris — COMMENTED** [review]
There is a failing build. Let's fix this first and then I will review the rest

File: pages/wizard/custom-primary-actions.page.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+    setActiveStepIndex(activeStepIndex - 1);
+  };
+
+  const onFinish = () => {
+    alert('Finish');
+  };
+
+  const customPrimaryActions = (
+    <SpaceBetween size="m" direction="horizontal">
```
Comment (just-boris) [inline pages/wizard/custom-primary-actions.page.tsx]: We should be using `xs` spacing size between buttons. There is a mention of this in our docs: https://cloudscape.design/components/button/?tabId=api

File: pages/wizard/custom-primary-actions.page.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      )}
+    </SpaceBetween>
+  );
+
+  return (
+    <Wizard
+      id="wizard"
+      steps={steps}
+      i18nStrings={i18nStrings}
+      allowSkipTo={true}
```
Comment (just-boris) [inline pages/wizard/custom-primary-actions.page.tsx]: What does this prop do in this case? if we render a fully custom actions, this property has no effect, right?

File: src/wizard/__tests__/wizard.test.tsx [hunk 3]
```diff
@@ -515,6 +515,83 @@ describe('Custom actions', () => {
...
+    });
+
+    DEFAULT_STEPS.forEach((_, index) => {
+      const customActionButtonWrapper = wrapper.findActions()!.findButton('[data-testid="custom-action"]');
+      expect(customActionButtonWrapper).not.toBeNull();
+      customActionButtonWrapper?.click();
+      expect(onCustomClick).toHaveBeenCalledTimes(index + 1);
+
+      // Navigate to next step for next iteration
+      console.log(wrapper.findContent()?.getElement().innerHTML);
```
Comment (just-boris) [inline src/wizard/__tests__/wizard.test.tsx]: We do not need console.logs in the mergeable code

File: src/wizard/__tests__/wizard.test.tsx [hunk 4]
```diff
@@ -515,6 +515,83 @@ describe('Custom actions', () => {
...
+
+    const [wrapper, rerender] = renderDefaultWizard({
+      customPrimaryActions: customActions,
+      activeStepIndex: 0,
+    });
+
+    DEFAULT_STEPS.forEach((_, index) => {
+      const customActionButtonWrapper = wrapper.findActions()!.findButton('[data-testid="custom-action"]');
+      expect(customActionButtonWrapper).not.toBeNull();
+      customActionButtonWrapper?.click();
```
Comment (just-boris) [inline src/wizard/__tests__/wizard.test.tsx]: This should be a strict non-null assertion, `!`. Because otherwise the test silently skips this line

File: src/wizard/__tests__/wizard.test.tsx [hunk 5]
```diff
@@ -515,6 +515,83 @@ describe('Custom actions', () => {
...
+    const [wrapper] = renderDefaultWizard({ customPrimaryActions: customActions });
+
+    expect(wrapper.findPrimaryButton()).toBeNull();
+    expect(wrapper.findCancelButton()).toBeNull();
+    expect(wrapper.findPreviousButton()).toBeNull();
+    expect(wrapper.findActions()!.findButton('[data-testid="custom-cancel"]')).not.toBeNull();
+    expect(wrapper.findActions()!.findButton('[data-testid="custom-next"]')).not.toBeNull();
+  });
+
+  test('custom primary actions work on all steps', () => {
```
Comment (just-boris) [inline src/wizard/__tests__/wizard.test.tsx]: Is it really needed to check on all steps? I think it would be better to check 3 corner-cases specifically 1) first step 2) any middle step 3) last step

File: src/wizard/interfaces.ts [hunk 6]
```diff
@@ -103,6 +103,13 @@ export interface WizardProps extends BaseComponentProps {
    */
   allowSkipTo?: boolean;
 
+  /**
+   * Specifies right-aligned custom primary actions for the wizard. Overwrites existing buttons (e.g. Cancel, Next, Finish, ...).
```
Comment (just-boris) [inline src/wizard/interfaces.ts]: This `...).` punctuation in the end looks strange, you will need to check with a designer if there is a better way

File: src/wizard/__tests__/wizard.test.tsx (hunk 5 above)
Comment (SpyZzey) [reply src/wizard/__tests__/wizard.test.tsx]: The DEFAULT_STEPS contains 3 steps, so it'll check the first, middle and last step or would you suggest separating the test into three separate ones?

File: src/wizard/__tests__/wizard.test.tsx (hunk 4 above)
Comment (SpyZzey) [reply src/wizard/__tests__/wizard.test.tsx]: The customActionButtonWrapper is asserted the line before, so it would throw an error before. But I'll fix it to be clearer.

File: src/wizard/__tests__/wizard.test.tsx (hunk 3 above)
Comment (SpyZzey) [reply src/wizard/__tests__/wizard.test.tsx]: Agree.

File: pages/wizard/custom-primary-actions.page.tsx (hunk 2 above)
Comment (SpyZzey) [reply pages/wizard/custom-primary-actions.page.tsx]: Correct. The property won't have an effect in this case. It's in the dev demo pages and I used it for manual testing.

File: src/wizard/__tests__/wizard.test.tsx (hunk 5 above)
Comment (just-boris) [reply src/wizard/__tests__/wizard.test.tsx]: yes, 3 separate ones would be easier to read and understand why they exist

**Review by just-boris — APPROVED** [review]
